source/code/plugins/oms_configuration.rb

```diff
@@ -147,7 +147,7 @@ def get_azure_resid_from_imds()
             if (imds_instance_json_compute['vmScaleSetName'].empty?)
               azure_resource_id = azure_resource_id + 'virtualMachines/' + imds_instance_json_compute['name']
             else
-              azure_resource_id = azure_resource_id + 'virtualMachineScaleSets/' + imds_instance_json_compute['vmScaleSetName'] + '/virtualMachines/' + imds_instance_json_compute['name']
+              azure_resource_id = azure_resource_id + 'virtualMachineScaleSets/' + imds_instance_json_compute['vmScaleSetName'] + '/virtualMachines/' + imds_instance_json_compute['name'].split('_')[-1]
```

Comment:
is there an example of how the instance name with "_" looks like in VMSS?

Reply:
![image](https://user-images.githubusercontent.com/43051891/57114511-6300ca80-6cfe-11e9-848c-706709d517a2.png)

Reply:
shouldn't the array index by 1 instead of -1 then?

Reply:
-1 gives the last element, 'name' is always the vmss name + '_' + instanceid, so -1 will always give us instanceId

Reply:
looks good